src/lib/macos-compat.c

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <unistd.h>
+#include <time.h>
+
+#include "macos-compat.h"
+
+// Time code based on: https://github.com/ChisholmKyle/PosixMachTiming/tree/master/src
```

Comment:
this is ISC-licensed, so bringing it into our tree would cause some licensing-related complications. i'll put together anything we need of this type. thanks for proving it out, though!